charms/worker/k8s/src/charm.py

```diff
@@ -197,6 +196,32 @@ def _apply_proxy_environment(self):
             with open("/etc/environment", mode="w", encoding="utf-8") as file:
                 file.write("\n".join([f"{k}={v}" for k, v in current_env.items()]))
 
+    def get_cluster_name(self) -> str:
+        """Craft a unique name for the cluster once joined or bootstrapped.
+
+        Note: It won't change for the lifetime of the unit.
+
+        Returns:
+            the cluster name.
+        """
+        unit, node = self.unit.name, self.get_node_name()
+        if not self._stored.cluster_name:
+            if self.lead_control_plane and self.api_manager.is_cluster_bootstrapped():
+                # TODO: replace with API call once available from the snap
+                self._stored.cluster_name = str(uuid.uuid4())
```

Comment:
Bootstrapped leader control-plane nodes can just make up a cluster name right now.  Something that is unique.  We should replace this by querying the kubernetes cluster itself -- but for now this is okay.